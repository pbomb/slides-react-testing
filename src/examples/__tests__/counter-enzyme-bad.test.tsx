import { mount, shallow } from 'enzyme';
import * as React from 'react';
import Counter, { Props, State } from '../counter';
import Display from '../display';

test('snapshots using mount', () => {
  expect(mount(<Counter />)).toMatchSnapshot();
});

test('snapshots using shallow', () => {
  expect(shallow(<Counter />)).toMatchSnapshot();
});

test('finding by components', () => {
  const wrapper = mount(<Counter />);

  const display = wrapper.find(Display);
  expect(display.text()).toBe('0');
});

test('reaching into component state', () => {
  const wrapper = mount<Props, State>(<Counter />, {});

  wrapper.find('[data-testid="increment"]').simulate('click');
  expect(wrapper.state().clicks).toBe(1);
});

test('reaching into component props', () => {
  const wrapper = mount(<Counter />);

  wrapper.find('[data-testid="increment"]').simulate('click');
  const display = wrapper.find(Display);

  expect(display.props().clicks).toBe(1);
});
