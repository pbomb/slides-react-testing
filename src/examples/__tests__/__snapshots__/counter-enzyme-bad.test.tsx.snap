// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots using mount 1`] = `
.emotion-0 {
  margin: 1em;
  color: black;
}

<Counter>
  <div>
    <button
      data-testid="decrement"
      disabled={false}
      onClick={[Function]}
    >
      -
    </button>
    <Component
      clicks={0}
    >
      <Styled(span)
        clicks={0}
        data-testid="count"
      >
        <span
          className="emotion-0"
          data-testid="count"
        >
          0
        </span>
      </Styled(span)>
    </Component>
    <button
      data-testid="increment"
      disabled={false}
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;

exports[`snapshots using shallow 1`] = `
<div>
  <button
    data-testid="decrement"
    disabled={false}
    onClick={[Function]}
  >
    -
  </button>
  <Component
    clicks={0}
  />
  <button
    data-testid="increment"
    disabled={false}
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
